tests/conftest.py

```diff
@@ -52,19 +53,29 @@ def pytest_configure(config):
     config.addinivalue_line("markers", "k8s: mark test to require a Kubernetes cluster")
 
     kubeconfig = config.getoption(KUBECONFIG_OPTION)
-    if kubeconfig:
+    k8s_context = config.getoption(KUBECONTEXT_OPTION)
+
+    if kubeconfig and not k8s_context:
         p = subprocess.run(
-            ["kubectl", "--kubeconfig", kubeconfig, "config", "current-context"],
+            [
+                "kubectl",
+                "--kubeconfig",
+                str(pathlib.Path(kubeconfig).expanduser()),
+                "config",
+                "current-context",
+            ],
             check=True,
             capture_output=True,
         )
         k8s_context = p.stdout.decode().strip()
 
-        if not k8s_context.startswith("crate-") and k8s_context != "minikube":
-            raise RuntimeError(
-                f"Cannot run tests on '{k8s_context}'. "
-                "Expected a context prefixed with 'crate-'."
-            )
+    if k8s_context and (
+        not k8s_context.startswith("crate-") and k8s_context != "minikube"
+    ):
```

Comment:
I don't think the condition here is not quite is correct.

When no `kubeconfig` is specified, integration tests will not be run at all, thus no check on the `k8s_context` is necessary. However, if it is defined, the `k8s_context` can either come from the CLI argument or, if that's not specified, from the `kubectl config current-context` command. If the `k8s_context` then is either empty or doesn't start with `crate-` or `is `minikube`, then we should raise the exception:

```python
def ...():
    kubeconfig = config.getoption(KUBECONFIG_OPTION)
    if kubeconfig:
        k8s_context = config.getoption(KUBECONTEXT_OPTION)
        if k8s_context is None:
            k8s_context = subprocess.run(...)
        if k8s_context is None or not k8s_context.startswith("crate-") and k8s_context != "minikube":
            raise
```

Reply:
As far as I tested it, the tests run if no `--kube-config` argument is passed. The other way to pass a config is by defining an env variable ( `KUBECONFIG`). As this is not documented in the development guide, I assume the default way for testing is the `--kube-config` argument?

So is it a bug that the tests are running if there is no `--kube-config` argument passed?

Reply:
Good point 👍